src/components/configurable-items/configurable-item-row/configurable-items-row.spec.js

```diff
@@ -0,0 +1,228 @@
+import Icon from '../../icon';
+import { WithDrag, WithDrop } from '../../drag-and-drop';
+import DraggableContext from '../../drag-and-drop/draggable-context';
+import { rootTagTest } from '../../../utils/helpers/tags/tags-specs';
+import { ConfigurableItemRowStyle, ConfigurableItemRowIconStyle } from './configurable-item-row.style';
+import classicTheme from '../../../style/themes/classic';
+import { assertStyleMatch } from '../../../__spec_helper__/test-utils';
+
+
+describe('ConfigurableItemRow', () => {
+  let wrapper;
+  const onChange = () => { };
+
+  describe('classNames', () => {
+    beforeEach(() => {
+      wrapper = shallow(<ConfigurableItemRowWithoutHOC className='my-custom-class-name' rowIndex={ 1 } />);
+      wrapper.instance().context = { dragAndDropActiveIndex: 1 };
+    });
+
+    it('adds the className to the row', () => {
+      expect(wrapper.find('.my-custom-class-name').length).toEqual(1);
+    });
+
+    describe('when the dragAndDropActiveIndex is the same as the rowIndex', () => {
+      beforeEach(() => {
+        wrapper.setProps({ rowIndex: 1 });
+      });
+
+      it('adds configurable-item-row--dragged to the classes', () => {
+        expect(wrapper.find(ConfigurableItemRowStyle).prop('isDragged')).toEqual(true);
+      });
+    });
+
+    describe('when the dragAndDropActiveIndex is the same as the rowIndex', () => {
+      beforeEach(() => {
+        wrapper.setProps({ rowIndex: 2 });
+      });
+      it('does not add configurable-item-row--dragged to the classes', () => {
+        expect(wrapper.find(ConfigurableItemRowStyle).prop('isDragged')).toEqual(false);
+      });
+    });
+  });
+
+  describe('tags', () => {
+    describe('on component', () => {
+      beforeEach(() => {
+        wrapper = shallow(
+          <ConfigurableItemRowWithoutHOC
+            data-element='bar' data-role='baz'
+            rowIndex={ 1 }
+          />
+        );
+      });
+      it('includes the correct component, element and role data tags', () => {
+        rootTagTest(wrapper, 'configurable-item-row', 'bar', 'baz');
+      });
+    });
+  });
+
+  describe('onChange', () => {
+    beforeEach(() => {
+      wrapper = shallow(
+        <ConfigurableItemRowWithoutHOC
+          onChange={ onChange } rowIndex={ 1 }
+        />
+      );
+    });
+    it('passes the onChange prop through to the Checkbox onChange prop', () => {
+      expect(wrapper.find(Checkbox).props().onChange).toEqual(onChange);
+    });
+
+    it('if classicTheme provided, render legacy checkbox component', () => {
+      wrapper.setProps({ theme: classicTheme });
+      expect(wrapper.find(CheckboxLegacy).props().onChange).toEqual(onChange);
+    });
+  });
+
+  describe('enabled', () => {
+    describe('when the row is enabled', () => {
+      beforeEach(() => {
+        wrapper = shallow(
+          <ConfigurableItemRowWithoutHOC enabled rowIndex={ 1 } />
+        );
+      });
+
+      it('sets the Checkbox value to true', () => {
```

Comment:
If the value here can be only `true` or `false` it would be better to test for that Ie. `toBe(true)` rather than risking a incorrect value being coerced to a boolean.

Reply:
fixed